### PR TITLE
Additional tests and fixes for dynamic config

### PR DESCRIPTION
**Pull request by vbekiaris**
Adds tests for `TopicConfig`, `SemaphoreConfig`, `ReliableTopicConfig` and `EventJournalConfig` along with small fixes. Improves coverage of some existing config tests.

Client-side event journal config requires https://github.com/hazelcast/hazelcast-client-protocol/pull/94

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/AddEventJournalConfigMessageTask.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Override
+    protected DynamicConfigAddEventJournalConfigCodec.RequestParameters decodeClientMessage(ClientMessage clientMessage) {
+        return DynamicConfigAddEventJournalConfigCodec.decodeRequest(clientMessage);
+    }
+
+    @Override
+    protected ClientMessage encodeResponse(Object response) {
+        return DynamicConfigAddSetConfigCodec.encodeResponse();
```
Comment (ihsandemir) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/AddEventJournalConfigMessageTask.java]: is this correct codec?

Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/dynamicconfig/AddEventJournalConfigMessageTask.java]: good catch, updated.

File: hazelcast-client/src/test/java/com/hazelcast/client/impl/ClientDynamicClusterConfigTest.java [hunk 2]
```diff
@@ -23,6 +23,8 @@
 import com.hazelcast.test.annotation.ParallelTest;
 import com.hazelcast.test.annotation.QuickTest;
 import org.junit.After;
+import org.junit.Ignore;
```
Comment (Donnerbart) [inline hazelcast-client/src/test/java/com/hazelcast/client/impl/ClientDynamicClusterConfigTest.java]: Those imports seem to be unused.

**Comment by vbekiaris** [comment]
run-lab-run

**Comment by vbekiaris** [comment]
verify

File: hazelcast-client/src/main/java/com/hazelcast/client/impl/ClientDynamicClusterConfig.java [hunk 3]
```diff
@@ -335,6 +338,20 @@ public Config addListenerConfig(ListenerConfig listenerConfig) {
     }
 
     @Override
+    public Config addEventJournalConfig(EventJournalConfig eventJournalConfig) {
+        String mapName = eventJournalConfig.getMapName();
+        String cacheName = eventJournalConfig.getCacheName();
+        if (StringUtil.isNullOrEmpty(mapName) && StringUtil.isNullOrEmpty(cacheName)) {
```
Comment (sancar) [inline hazelcast-client/src/main/java/com/hazelcast/client/impl/ClientDynamicClusterConfig.java]: Isn't one of them being not null was enough ? This look like it should have been ||, right?
And the message is confusing in exception. 
`Event journal config should have non-empty map name and/or cache name`
Is it `and` or `or` ? According to code it should be just `and`

**Pull request merged**